cygnus-common/src/main/java/com/telefonica/iot/cygnus/management/MetricsHandlers.java

```diff
@@ -0,0 +1,213 @@
+     * @param sources
+     * @param sinks
+     * @throws java.io.IOException
+     */
+    public static synchronized void delete(HttpServletResponse response, ImmutableMap<String, SourceRunner> sources,
+            ImmutableMap<String, SinkRunner> sinks) throws IOException {
+        deleteMetrics(sources, sinks);
+        response.setStatus(HttpServletResponse.SC_OK);
+    } // delete
+    
+    /**
+     * Merges metrics from all the given sources and sinks. It is protected in order it can be tested.
+     * @param sources
+     * @param sinks
+     * @return
+     */
+    protected static CygnusMetrics mergeMetrics(ImmutableMap<String, SourceRunner> sources,
+            ImmutableMap<String, SinkRunner> sinks) {
+        CygnusMetrics mergeServices = new CygnusMetrics();
+    
+        if (sources != null) {
+            for (String key : sources.keySet()) {
+                Source source;
+                HTTPSourceHandler handler;
+
+                try {
+                    SourceRunner sr = sources.get(key);
+                    source = sr.getSource();
+                    Field f = source.getClass().getDeclaredField("handler");
+                    f.setAccessible(true);
+                    handler = (HTTPSourceHandler) f.get(source);
+                } catch (IllegalArgumentException | IllegalAccessException | NoSuchFieldException
+                        | SecurityException e) {
+                    LOGGER.error("There was a problem when getting a sink. Details: " + e.getMessage());
+                    continue;
+                } // try catch
+
+                if (handler instanceof CygnusHandler) {
+                    CygnusHandler ch = (CygnusHandler) handler;
+                    CygnusMetrics sourceServices = ch.getServiceMetrics();
+                    mergeServices.merge(sourceServices);
+                } // if
+            } // for
+        } // if
+        
+        if (sinks != null) {
+            for (String key : sinks.keySet()) {
+                Sink sink;
+
+                try {
+                    SinkRunner sr = sinks.get(key);
+                    SinkProcessor sp = sr.getPolicy();
+                    Field f = sp.getClass().getDeclaredField("sink");
+                    f.setAccessible(true);
+                    sink = (Sink) f.get(sp);
+                } catch (IllegalArgumentException | IllegalAccessException | NoSuchFieldException
+                        | SecurityException e) {
+                    LOGGER.error("There was a problem when getting a sink. Details: " + e.getMessage());
+                    continue;
+                } // try catch
+
+                if (sink instanceof CygnusSink) {
+                    CygnusSink cs = (CygnusSink) sink;
+                    CygnusMetrics sinkServices = cs.getServiceMetrics();
+                    mergeServices.merge(sinkServices);
+                } // if
+            } // for
+        } // if
+
+        return mergeServices;
+    } // mergeServices
```

Comment:
Fixed in https://github.com/telefonicaid/fiware-cygnus/pull/1383/commits/1ca87236b3c2b0b44789f1a3d2cf6801be79fb8d